checker/src/fixed_point_visitor.rs

```diff
@@ -242,6 +242,27 @@ impl<'fixed, 'analysis, 'compilation, 'tcx, E>
             .0;
 
         let at_loop_anchor = self.loop_anchors.contains(&bb);
+
+        // Identifies loop invariants. Currently we only extract invariant paths, i.e., paths that are not
+        // mutated by the loop. todo: support more kinds of invariants.
+        let mut joined_state_without_loop_variants = joined_state.clone();
+        if at_loop_anchor {
+            self.bv.mir.predecessors()[bb].iter().for_each(|pred_bb| {
+                if self.dominators.is_dominated_by(*pred_bb, bb) {
+                    let pred_out_state = &self.out_state[pred_bb];
+                    for (path, val) in pred_out_state.value_map.iter() {
+                        if let Some(widened_val) = joined_state.value_at(path) {
```

Comment:
This value may not be widened, so the name is not that helpful. I had to read this a second time because of it.

Reply:
Renamed to `joined_val`.